libcaf_core/caf/flow/op/merge.hpp

```diff
@@ -87,6 +87,8 @@ class merge_sub : public subscription::impl_base,
 
   void on_next(const observable<T>& what) override {
     CAF_ASSERT(what);
+    if (!sub_)
+      return;
```

Comment:
Shouldn't we add the same guard to `on_complete` and `on_error`?

Reply:
I have not completed the testing yet and there are some scenarios I am trying to check. But, both of `on_complete` and `on_error` resets sub_ to nullptr. I am not sure if it is necessary yet.

Reply:
I have added a check for `out_` in `on_error`. But, as we are not doing any operation in sub_ and setting it to nullptr, I don't think it is required to add `if (!sub_)` check in `on_error` and `on_complete`.